src/cheerio_crawler.js

```diff
@@ -153,15 +183,21 @@ class CheerioCrawler {
             maxRequestsPerCrawl,
             handleFailedRequestFunction,
             autoscaledPoolOptions,
-        } = _.defaults(options, DEFAULT_OPTIONS);
+        } = _.defaults({}, options, DEFAULT_OPTIONS);
 
         checkParamOrThrow(handlePageFunction, 'options.handlePageFunction', 'Function');
         checkParamOrThrow(requestFunction, 'options.requestFunction', 'Maybe Function');
+        checkParamOrThrow(requestOptions, 'options.requestOptions', 'Maybe Object');
         checkParamOrThrow(requestTimeoutSecs, 'options.requestTimeoutSecs', 'Number');
         checkParamOrThrow(handlePageTimeoutSecs, 'options.handlePageTimeoutSecs', 'Number');
-        checkParamOrThrow(ignoreSslErrors, 'options.ignoreSslErrors', 'Boolean');
-
-        this.ignoreSslErrors = ignoreSslErrors;
+        checkParamOrThrow(ignoreSslErrors, 'options.ignoreSslErrors', 'Maybe Boolean');
+        checkParamOrThrow(useApifyProxy, 'options.useApifyProxy', 'Maybe Boolean');
```

Comment:
In a case of booleans might be better to have type enforced to `Boolean` and set default value either in DEFAULT_OPTIONS or in destructuring like

```javascript
const {
    ...
    useApifyProxy = false,
    ...
} = _.defaults({}, ...);
```